### PR TITLE
Simplify the codebase and math

### PR DESCRIPTION
**Pull request by jokke-ilujo**
This change leaves the original code in comments and tries to explain the reasoning for changes under NOTE(jokke): lines.

### CODE REVIEW

**Comment by digimer** [comment]
Hi Jokke! I've merged a few PRs, could I trouble you to resolve the conflicts? I want to make sure your contribution is properly merged in. Thanks!!

**Comment by jokke-ilujo** [comment]
Hello, sorry for the late reply, last weekend was FOSDEM travels and the week after been bit busy.

I looked the merge conflict briefly and as the changes pretty much removed like 3rd of the lines, moved around or touched otherwise about half what was left it looks like I'd need to redo/alter most of what I did there. Honestly I don't think that's productive use of my time. The first commit in the chain has comments on pretty much everything I did, it shouldn't be hard to follow. Feel free to solve the conflicts, pick and choose parts of what I did that still fits, pass it to someone else to learn and sort out or just drop it. While I appreciate your thought, I really don't care about the attribution enough to redo the work.

There is no hard feelings if you don't utilize the PR, I'll keep my eye on the repo every now and then and you might see other ones from me.

**Comment by digimer** [comment]
No worries at all; I switched to getting the motor connected and the last video out, so I was holding on more code work myself. I'll be testing this now.

**Comment by digimer** [comment]
Thank you very much for this, @jokke-ilujo ! I'm sorry I didn't merge, I've been on quite the learning curve. I just finished a heavy refactor with PR 21, so I am going to close this. Again, I very much appreciate your contribution!!
